### PR TITLE
#1920 chart detail selector popup

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

Custom values: Helm chart detail- popup, listing and navigation 

Fixes # [(issue)](https://github.com/devtron-labs/devtron/issues/1920)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Manually tested

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by sohel-rp — APPROVED** [review]

File: src/components/charts/discoverChartDetail/types.tsx [hunk 1]
```diff
@@ -1,19 +1,71 @@
...
+    closePopup: () => void
+    appStoreApplicationName: string
+    appIconUrl: string
+    onError: (e) => void
+    handleDeploy: () => void
+    deployedChartValueList: ChartValuesType[]
+    presetChartValueList: ChartValuesType[]
+}
 
+export const ValueType = {
```
Comment (shivani170) [inline src/components/charts/discoverChartDetail/types.tsx:40]: this function must come utils instead type

File: src/components/charts/discoverChartDetail/ChartVersionSelectorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            >
+                <div className={`header-container mt-20 mr-20 mb-20 ${isListpage ? '' : 'ml-20'}`}>
+                    {isListpage ? renderListHeader() : renderInitialHeader()}
+                    <button type="button" className="transparent" onClick={closePopup}>
+                        <Close className="icon-dim-20" />
+                    </button>
+                </div>
+                {isListpage ? renderValueList() : renderValueTypeList()}
+                {isListpage && (
+                    <div className="pt-20 pr-20 pb-20 pl-20 border-top right-align">
```
Comment (shivani170) [inline src/components/charts/discoverChartDetail/ChartVersionSelectorModal.tsx:194]: Can isolate it as a functional component.

File: src/components/charts/discoverChartDetail/types.tsx [hunk 3]
```diff
@@ -1,19 +1,71 @@
...
-    appStoreApplicationName?: string;
-    deprecated: boolean;
-    isGitOpsConfigAvailable;
-    showGitOpsWarningModal;
-    toggleGitOpsWarningModal;
-    availableVersions: Map<number, { id, version }>;
-}
\ No newline at end of file
+export const PrimaryOptions: PrimaryOptionType[] = [
+    {
```
Comment (shivani170) [inline src/components/charts/discoverChartDetail/types.tsx:47]: Same

File: src/setupProxy.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ module.exports = function (app) {
     app.use(
```
Comment (shivani170) [inline src/setupProxy.js:4]: are we committing set up proxy as well?

Comment (vivek-devtron) [reply src/setupProxy.js:4]: It is for this rel branch won't be committing it into main

File: src/components/charts/discoverChartDetail/ChartVersionSelectorModal.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/charts/discoverChartDetail/ChartVersionSelectorModal.tsx]: done

File: src/components/charts/discoverChartDetail/types.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/components/charts/discoverChartDetail/types.tsx]: done

**Review by shivani170 — APPROVED** [review]

**Pull request merged**